File: front/src/entity/z3950.ts
export class Z3950 {
  private id: number;
  private name: string;
  private url: string;
  private port: number;
  private syntax: string;
  private database: Map<string, string>;
  private options: Map<string, string>;

  getId(): number {
    return this.id;
  }

  setId(value: number) {
    this.id = value;
  }

  getName(): string {
    return this.name;
  }

  setName(value: string) {
    this.name = value;
  }

  getUrl(): string {
    return this.url;
  }

  setUrl(value: string) {
    this.url = value;
  }

  getPort(): number {
    return this.port;
  }

  setPort(value: number) {
    this.port = value;
  }

  getSyntax(): string {
    return this.syntax;
  }

  setSyntax(value: string) {
    this.syntax = value;
  }

  getDatabase(): Map<string, string> {
    return this.database;
  }

  setDatabase(value: Map<string, string>) {
    this.database = value;
  }

  getOptions(): Map<string, string> {
    return this.options;
  }

  setOptions(value: Map<string, string>) {
    this.options = value;
  }
}

File: back/src/main/java/org/lendi/umtapo/entity/configuration/Z3950.java
package org.lendi.umtapo.entity.configuration;

import java.util.Map;

/**
 * Z39.50 entity.
 */
public class Z3950 {
    private int id;
    private String name;
    private String url;
    private int port;
    private String syntax;
    /**
     * Time to live connection in ms.
     */
    private long ttl;
    private Map<String, String> database;
    private Map<String, String> options;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public int getPort() {
        return port;
    }

    public void setPort(int port) {
        this.port = port;
    }

    public String getSyntax() {
        return syntax;
    }

    public void setSyntax(String syntax) {
        this.syntax = syntax;
    }

    public long getTtl() {
        return ttl;
    }

    public void setTtl(long ttl) {
        this.ttl = ttl;
    }

    public Map<String, String> getDatabase() {
        return database;
    }

    public void setDatabase(Map<String, String> database) {
        this.database = database;
    }

    public Map<String, String> getOptions() {
        return options;
    }

    public void setOptions(Map<String, String> options) {
        this.options = options;
    }
}
